packages/presentation/lib/screens/home/home_bloc.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:domain/entities/base_movie_entity.dart';
+import 'package:domain/usecases/convert_api_runtime_usecase.dart';
+import 'package:domain/usecases/get_image_url_usecase.dart';
+import 'package:get_it/get_it.dart';
+import 'package:domain/usecases/get_coming_soon_movies_usecase.dart';
+import 'package:domain/usecases/get_now_showing_movies_usecase.dart';
+import 'package:presentation/bloc/base/bloc.dart';
+import 'package:presentation/bloc/base/bloc_impl.dart';
+import 'package:presentation/screens/home/data/home_data.dart';
+import 'package:presentation/screens/movie_details/movie_details_screen.dart';
+
+abstract class HomeBloc implements Bloc<HomeData?> {
+  factory HomeBloc() => _HomeBloc(
+        GetIt.I.get<GetNowShowingMoviesUseCase>(),
+        GetIt.I.get<GetComingSoonMoviesUseCase>(),
+        GetIt.I.get<ConvertApiRuntimeUsecase>(),
+        GetIt.I.get<GetImageUrlUseCase>(),
+      );
+
+  Future<void> showNowShowingMovies();
+  Future<void> showComingSoonMovies();
+  String convertApiRuntime(int runtime);
+  String? getImageUrlById(String? id);
+  void goToMovieDetailsPage(BaseMovieEntity movieDetails);
+}
+
+class _HomeBloc extends BlocImpl<HomeData?> implements HomeBloc {
+  final GetNowShowingMoviesUseCase _getNowShowingMoviesUseCase;
+  final GetComingSoonMoviesUseCase _getComingSoonMoviesUseCase;
+  final ConvertApiRuntimeUsecase _convertApiRuntimeUsecase;
+  final GetImageUrlUseCase _getImageUrlUseCase;
+
+  _HomeBloc(
+    this._getNowShowingMoviesUseCase,
+    this._getComingSoonMoviesUseCase,
+    this._convertApiRuntimeUsecase,
+    this._getImageUrlUseCase,
+  ) : super(initState: const HomeData.init());
+
+  @override
+  void initState() {
+    showNowShowingMovies();
+  }
+
+  @override
+  Future<void> showNowShowingMovies() async {
+    add(null);
+    final movies = await _getNowShowingMoviesUseCase();
+    _updateHomeDataWithMovies(movies);
+  }
+
+  @override
+  Future<void> showComingSoonMovies() async {
+    add(null);
+    final movies = await _getComingSoonMoviesUseCase();
+    _updateHomeDataWithMovies(movies);
+  }
+
+  void _updateHomeDataWithMovies(List<BaseMovieEntity> movies) {
+    final newState = HomeData(movies);
+    add(newState);
+  }
+
+  @override
+  String convertApiRuntime(int runtime) {
+    return _convertApiRuntimeUsecase(runtime);
```

Comment:
Why u need convertApiRuntime usecase instead of converting this data on domain layer?

Reply:
I did it because at first grid shows only 4 widgets for which there are 4 conversions and next conversions can happen if user scroll down grid widget, if I did it in domain layer I had to create about 60 conversions, this is the main reason.

Reply:
probably don't agree. your list will receive already mapped data. mapping will be done before the displaying list

Reply:
Fixed